doc/spec/namespace.md

```diff
@@ -0,0 +1,198 @@
+characters however there is no specific limitation on DNS or path components.
+
+### Name Grammar
+```
+<name> ::= <hostname>"/"<path>
+<hostname> ::= <host-part>*["."<host-part>][":"<port>]
+<path> ::= <path-part>*["/"<path-part>]
+<host-part> ::= <regexp "[a-z]([-]?[a-z0-9])*">
+<port> ::= <number 1 to 65535>
+<path-part> ::= <regexp "[a-z0-9]([._-]?[a-z0-9])*">
+```
+
+## Metadata
+The metadata for a namespce is a list of entries consisting of a scope, action,
+and space separate arguments. Each action may interpret the arguments
+differently. The scope of an individual entry defines which namespaces the
+value may apply. It is up to the resolution process to return the set of
+metadata which should be applied for a given name, however any returned values
+which are out of scope should be considered invalid.
+
+### Actions
+#### pull
+
+Used to represent a registry endpoint which supports pull operations. This
+may include full registries as well as read-only mirrors.
+
+The arguments for pull consist of a registry endpoint as well as optional
+arguments for priority and key=value flags.
+
+`<registry endpoint> [<priority>] [<flag>[=<value>], ...]`
+
+##### Priority
+
+Integer value providing relative sort order between other endpoints with the
+same action. Higher priority endpoints should be tried before lower priority
+endpoints.
+
+##### Flags
+
+| Key | Value | Default | Description |
+|---|---|---|---|
+| trim | boolean | false | Whether this registry endpoint expects the hostname to be trimmed from the api requests. This is used for compatibility with existing registries |
+| version | string | "2.0" | Which API version the registry implements |
+| notag | boolean | false | Whether tag operations are not supported by this registry |
+
+#### push
+
+Used to represent a registry endpoint which supports push operations. Should
+never be defined for read-only mirrors.
+
+Push uses the same arguments as pull.
+
+#### index
+
+Used to represent a search index endpoint.
+
+`<registry endpoint> [version=<value>]`
+
+#### namespace
+
+Used to extend the interface to a parent or stop further namespace processing
+by not providing any arguments. When a parent is provided as an argument,
+namespace processing should continue by including the resolved values of the
+the parent. A namespace action without a scope can be used to turn off a
+namespace by providing no values except a namespace action. This will end
+processing since a value is found however not provide any metadata which can
+be used to configure the endpoint.
+
+`[<parent scope>, ...]`
+
+## Discovery
+The discovery process involves resolving a name into namespace scoped metadata.
+The namespace metadata contains the full set of information needed to fetch and
+verify content associated with the namespace repository. The metadata includes
+list of registry api endpoints, the trust model, and search index. The discovery
```

Comment:
Using HTTPS does provide a reasonable amount of confidence in the content, but it will not be only method for discovering the data. The meaning here is that the information provided through discovery should be able to have additional methods of verification, whether key pinning with trust on first use, certificate chain, or whitelisting. I am not defining the method for trusting the information, only the transport for getting it. Relying entirely on HTTPS could certainly be a model to trust the information.